pkg/controller/che/k8s_helpers.go

```diff
@@ -257,23 +256,24 @@ func (cl *k8s) GetDeploymentPod(name string, ns string) (podName string, err err
 // There's an easier way which is to read tls secret in default (3.11) or openshift-ingress (4.0) namespace
 // which however requires extra privileges for operator service account
 func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, url string) (certificate []byte, err error) {
-	testRoute := &routev1.Route{}
 	var requestURL string
+	var testRoute *routev1.Route
 	if len(url) < 1 {
-		testRoute = deploy.NewTlsRoute(instance, "test", "test", 8080)
-		logrus.Infof("Creating a test route %s to extract router crt", testRoute.Name)
-		if err := r.CreateNewRoute(instance, testRoute); err != nil {
-			logrus.Errorf("Failed to create test route %s: %s", testRoute.Name, err)
-			return nil, err
+		testRoute = r.GetEffectiveRoute(instance, "test")
+		if testRoute == nil {
+			testRoute = deploy.NewTlsRoute(instance, "test", "test", 8080)
+			logrus.Infof("Creating a test route %s to extract router crt", testRoute.Name)
+			if err := r.CreateNewRoute(instance, testRoute); err != nil {
+				logrus.Errorf("Failed to create test route %s: %s", testRoute.Name, err)
+				return nil, err
+			}
 		}
-		// sometimes timing conditions apply, and host isn't available right away
+
 		if len(testRoute.Spec.Host) < 1 {
```

Comment:
So here it seems you'll always be in this case the first time since you don't get the effective route object after creating it in line [266](https://github.com/eclipse/che-operator/pull/66/files#diff-5899627ff2ee9c6f7afb34d27f2d3b3eR266) ?

Wouldn't it be a good idea to add

```go
testRoute = r.GetEffectiveRoute(instance, "test")
```

also just after line [266](https://github.com/eclipse/che-operator/pull/66/files#diff-5899627ff2ee9c6f7afb34d27f2d3b3eR266) ?

Reply:
I took a quick look on it and checked that 
```
r.client.Create(context.TODO(), route)
```
store the created object(response from server) into the second argument
See
https://github.com/eclipse/che-operator/blob/bc59de083a8be1bb3cdf53110bafc82cc23d3f84/vendor/sigs.k8s.io/controller-runtime/pkg/client/client.go#L103
https://github.com/eclipse/che-operator/blob/bc59de083a8be1bb3cdf53110bafc82cc23d3f84/vendor/sigs.k8s.io/controller-runtime/pkg/client/typed_client.go#L44
https://github.com/eclipse/che-operator/blob/bc59de083a8be1bb3cdf53110bafc82cc23d3f84/vendor/k8s.io/client-go/rest/request.go#L1094

Let me take a look closer and test it one more time

Reply:
Thanks for checking !

Reply:
Thanks for checking !

Reply:
![Screenshot_20190905_112140](https://user-images.githubusercontent.com/5887312/64324630-5c959e80-cfcf-11e9-9b51-eab6b67354cc.png)
As we can see here - route host is fetched from the first try

Reply:
Great, Thanks Sergii !